core/src/main/scala/org/broadinstitute/dsde/rawls/snapshot/SnapshotService.scala

```diff
@@ -56,50 +57,62 @@ class SnapshotService(protected val ctx: RawlsRequestContext,
     with WorkspaceSupport
     with LazyLogging {
 
-  def createSnapshot(workspaceName: WorkspaceName,
-                     snapshotIdentifiers: NamedDataRepoSnapshot
+  // Finds a workspace using the workspaceId then calls the createSnapshot method
+  def createSnapshotByWorkspaceId(workspaceId: String,
+                                  snapshotIdentifiers: NamedDataRepoSnapshot
+  ): Future[DataRepoSnapshotResource] =
+    getV2WorkspaceContextAndPermissionsById(workspaceId, SamWorkspaceActions.write).flatMap { rawlsWorkspace =>
```

Comment:
last step on the `attributeSpecs` saga: specify here that you don't care to retrieve any attributes. Note this is parallel to the behavior in `createSnapshotByWorkspaceName`:
```suggestion
    getV2WorkspaceContextAndPermissionsById(workspaceId,
                                            SamWorkspaceActions.write,
                                            Some(WorkspaceAttributeSpecs(all = false))
    ).flatMap { rawlsWorkspace =>
```